adapters/feeder2core/feeder2core.go

```diff
@@ -245,14 +246,58 @@ func AdaptCairo1Class(response *feeder.SierraDefinition, compiledClass json.RawM
 		class.EntryPoints.Constructor[index] = core.SierraEntryPoint{Index: v.Index, Selector: v.Selector}
 	}
 
-	if compiledClass != nil {
-		if err = json.Unmarshal(compiledClass, &class.Compiled); err != nil {
-			return nil, err
-		}
+	if class.Compiled, err = AdaptCompiledClass(compiledClass); err != nil {
+		return nil, err
 	}
+
 	return class, nil
 }
 
+func AdaptCompiledClass(compiledClass *feeder.CompiledClass) (core.CompiledClass, error) {
+	compiled := core.CompiledClass{}
+	compiled.Bytecode = compiledClass.Bytecode
+	pyHints, err := json.Marshal(compiledClass.PythonicHints)
+	if err != nil {
+		return core.CompiledClass{}, err
+	}
+	compiled.PythonicHints = pyHints
+	compiled.CompilerVersion = compiledClass.CompilerVersion
+	var success bool
+	compiled.Prime, success = new(big.Int).SetString(compiledClass.Prime, 0)
+	if !success {
+		return core.CompiledClass{}, fmt.Errorf("couldn't convert prime value to big.Int")
```

Comment:
```suggestion
		return core.CompiledClass{}, fmt.Errorf("couldn't convert prime value to big.Int: %d", compiled.Prime)
```
or something like that